docs/sources/tempo/operations/best-practices.md

```diff
@@ -0,0 +1,76 @@
+
+For more information, read the [Attribute and Resource](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md) sections in the OpenTelemetry specification.
+
+
+### Naming conventions for span and resource attributes
+
+When naming attributes, it is best to use consistent, nested namespaces.
+This ensures that attribute keys will be obvious to anyone observing the spans of a trace, and that common attributes can be shared by spans.
+Using our example from above, the `http` prefix of the attribute is the namespace, with `url` and `status_code` being keys within that namespace.
+These can also be nested, for example `http.url.protocol` might be `HTTP` or `HTTPS`, whereas `http.url.path` might be `/api/v1/query`.
+
+There are more details around semantic naming conventions which should be followed at the following link: https://opentelemetry.io/docs/specs/otel/common/attribute-naming/#recommendations-for-opentelemetry-authors
+
+Some third-party libraries already provide auto-instrumentation that generate span and span attributes when included in a source base.
+This alleviates the need for you to add spans and attributes for calling those libraries.
+
+For more information about instrumenting your app for tracing, refer to the [Instrument for distributed tracing](/docs/tempo/latest/getting-started/instrumentation/) documentation.
+
+
+## Determining where to add spans
+
+Spans make up a trace, where a trace is essentially just a meta ID that groups spans together.
+Spans themselves denote units of work. This could be something that carries out some work within a service, or it could be a call from, or to, another service that is upstream or downstream.
+Spans also have a hierarchy, where parent spans can have children or siblings.
+
+When instrumenting, determine the smallest piece of work that you need to observe in a trace to be of value to ensure that you don’t over (or under) instrument.
+
+In general, when manually instrumenting, create a new span for any work that has a relatively significant duration. This allows the observation of a trace to immediately show where significant amounts of time are spent during the processing of a request into your application or system.
+
+For example, adding a span for a call to another services (either instrumented or not) may take an unknown amount of time to complete, and therefore being able to separate this work shows when services are taking longer than expected.
+
+Adding a span for a piece of work that might call many other functions in a loop is a good signal of how long that loop is taking (you might add a span attribute that counts how many time the loop runs to determine if the duration is acceptable).
+However, adding a span for each method or function call in that loop might not, as it might produce hundreds or thousands of spans that are essentially of no individual value.
```

Comment:
```suggestion
When instrumenting, determine the smallest piece of work that you need to observe in a trace to be of value to ensure that you don’t over (or under) instrument.

In general, when manually instrumenting, create a new span for any work that has a relatively significant duration. This allows the observation of a trace to immediately show where significant amounts of time are spent during the processing of a request into your application or system.

For example, adding a span for a call to another services (either instrumented or not) may take an unknown amount of time to complete, and therefore being able to separate this work shows when services are taking longer than expected.

Adding a span for a piece of work that might call many other functions in a loop is a good signal of how long that loop is taking (you might add a span attribute that counts how many time the loop runs to determine if the duration is acceptable).
However, adding a span for each method or function call in that loop might not, as it might produce hundreds or thousands of spans that are essentially of no individual value.
```